apps/src/templates/sectionProgress/progressTables/ProgressTableView.jsx

```diff
@@ -271,12 +316,17 @@ class ProgressTableView extends React.Component {
       ? this.detailContentViewProps()
       : this.summaryContentViewProps();
 
+    // we use the view type as a key to force a full re-instantiation of the
+    // table components when the view changes
+    const key = this.props.currentView;
+
     return (
       // outer div contains both table and legend
       <div>
         <div style={styles.container} className="progress-table">
           <div style={styles.studentList} className="student-list">
             <ProgressTableStudentList
+              key={key}
```

Comment:
re-instantiating the content view without re-instantiating the student list was at the core of the problems i was seeing -- not only did that mean resetting the vertical scroll of the content view and not the student list, but also resulted in their internal states being out of sync.